# Outline of Steps

Plan: go through research paper and outline steps

https://www.biorxiv.org/content/10.1101/2020.05.13.094854v1.full


Abstract (might be later in the process)




<h3> Summary of abstract section </h3>

One way to investigate cellular responses to injury is through transcriptomic profiling of cell types that comprise the injury site.


1) Used sc-RNAseq generate a single cell transcriptomics dataset of virtually all cell types that comprise the uninjured and injured spinal cord at 1, 3, and 7dpi

2) From this dataset, we were able to obtain unique molecular signatures of multiple cell types as well as their subpopulations present throughout the acute injury phase. 

3) By assessing expression of ligandreceptor pairs on different cell types, we were able to gain insight into potential signaling relationships that mediat angiogenesis, gliosis, and fibrosis.


<h3> Summary of results section </h3>

1) To assess the cellular heterogeneity among all cell populations at the injury site, we obtained a total of 51,843 cells from uninjured and 1, 3, and 7dpi tissue, which resulted in a total of 15 distinct clusters when visualized on a UMAP plot (Fig. 1a, b).

2) These 15 clusters represented all major cell types that are known to comprise the SCI site including microglia, monocytes, macrophages, neutrophils, dendritic cells, astrocytes, oligodendrocytes, OPCs, neurons, fibroblasts, pericytes, ependymal cells, and endothelial cells. These cell types were grouped into three categories, namely myeloid, vascular, and macroglia, for further analysis as described below.

3) Neurons were excluded from the analysis because they were not expected to survive our dissociation protocol and thus subject to a selection bias.

4) Lymphocytes were also excluded because they are primarily involved in autoimmunity after SCI10, which is beyond the scope of this study. Cell types pertaining to each cluster were identified using annotated lineage markers (Extended Data Fig. 1).

5) The highest differentially expressed genes (DEGs) provided a unique molecular signature for each cell type (Fig. 1c, d), which in most cases were different from canonical markers used in the literature.

6) For example, the highest DEGs in OPCs were non-canonical genes such as tnr (Tenascin-R) and lhfpl3 (lipoma HMGIG fusion partner), which displayed better specificity than canonical OPC markers such as pdgfra and cspg4 that were expressed in multiple cell types (Extended Data Fig. 1).

7)  Interestingly, while certain marker genes were expressed both before and after injury, others such as postn in fibroblasts, changed expression in an injury-dependent manner. 

8) While postn was not expressed in uninjured fibroblasts, there was a graded increase as injury progressed (Fig. 1c). We validated this by genetic lineage tracing in PostnEYFP mice, which showed EYFP cells present in the fibrotic scar and overlying meninges but absent in surrounding spinal cord tissue. (Extended Data Fig. 2). 

9) Taken together, our analysis of DEGs between major cell types uncovers highly specific molecular identifiers, many of which are non-canonical and display temporal specificity.


<h3> Summary Outline for "Myeloid analysis reveals temporal changes in macrophage and microglial subtypes" </h3>
